pkg/inputs/http/http.go

```diff
@@ -153,6 +154,34 @@ func (ks *KentikHttpListener) getJCHF(wrapper *basic, remoteIP string) *kt.JCHF
 	return in
 }
 
+// Get the JCHF content directly.
+func (ks *KentikHttpListener) readJCHF(w http.ResponseWriter, r *http.Request) {
+	wrapper := []*kt.JCHF{}
+
+	// Decode body in gzip format if the request header is set this way.
+	body := r.Body
+	if r.Header.Get("Content-Encoding") == "gzip" {
+		z, err := gzip.NewReader(r.Body)
+		if err != nil {
+			panic(http.StatusInternalServerError)
```

Comment:
Any reason not to log as Error and return to keep running in case of a temporary fail instead of panic?

Reply:
This is some hipster stuff I stole from @wg . These handlers are wrapped in `wrap()` here: https://github.com/kentik/ktranslate/blob/481a76e2ed93475b61bde46308009d7ff1755b04/pkg/inputs/http/http.go#L195

Uses the panic/recovery method to pass the code back to the caller but keep going. At scale I'm not sure how performant it is but it helps to simplify the code.